pkgs/development/python-modules/nimbo/default.nix

```diff
@@ -0,0 +1,35 @@
+{ lib, python3, fetchFromGitHub, awscli }:
+python3.pkgs.buildPythonApplication rec {
```

Comment:
Using `pythonXPackages` isn't allowed from `python-modules` as this will mix interpreter versions for a given package. For example, if you were to do `python37Packages.<mypkg>`, but the expression pulled from `python3[8]Packages`. Python will be mis-importing several libraries, and this will likely cause issues at runtime.

Please refer to [Python dependency handling](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#handling-dependencies) on the best practices for handling dependencies within `python-modules`. Namely, you should be importing each python package individually, and not refer to another python package set (E.g. `python3Packages`).

If you're new to nixpkgs, you're free to view some of my videos on nixpkgs and python:
- [Python packaging using nixpkgs](https://www.youtube.com/watch?v=jXd-hkP4xnU)
- [Adding a package to nixpkgs unstable](https://www.youtube.com/watch?v=fvj8H5yUKu8)

Reply:
I looked at how `black` did this and tried to copy it, d4ff11c9567cb2b131240d664fd1d52fe0154d83, sorry for getting this wrong twice

Reply:
black is packaged outside of `python-modules`. Its a pythonApplication, which gets treated differently

Reply:
Now I'm confused then, I'll try and explain what I understand to be true and hopefully you can tell me where I'm stupid.
The [official docs](https://nixos.org/manual/nixpkgs/stable/#contributing) say:
> - Python libraries are called from python-packages.nix and packaged with buildPythonPackage. The expression of a library should be in pkgs/development/python-modules/<name>/default.nix. Libraries in pkgs/top-level/python-packages.nix are sorted quasi-alphabetically to avoid merge conflicts.
> - Python applications live outside of python-packages.nix and are packaged with buildPythonApplication.
(from (here)[Python libraries are called from python-packages.nix and packaged with buildPythonPackage. The expression of a library should be in pkgs/development/python-modules/<name>/default.nix. Libraries in pkgs/top-level/python-packages.nix are sorted quasi-alphabetically to avoid merge conflicts.

Then as nimbo is a python application, it should live outside of `python-packages.nix` i.e. in `all-packages.nix`, but if I'm importing all the python packages individually then the only place that it works is if the `callPackage` call is in `python-packages.nix`? I looked at how black did this and it is defined in `python-packages.nix` _and_ `all-packages.nix` with:
```
  black = with python3Packages; toPythonApplication black;
```
Which is what I changed nimbo to do in the latest commit. Also, I'm extra confused because I can only find black at the path `./pkgs/development/python-modules/black`, I don't understand what you mean that it's packaged outside of `python-modules`. I only switched to the `with python3.pkgs` method of packaging on the recommendation of @fabaff in the previous PR #121061, I'm only trying to follow the advice given to me by the docs and members here, but it's proving impossible to follow them all, I don't know what's correct anymore 😰.

Reply:
> black is packaged outside of `python-modules`. Its a pythonApplication, which gets treated differently

Yea, this is wrong, it's in `python-modules`, just also exported with `toPythonApplication`.

More succinctly, if people are expected to do `import nimbo`, then it should be in `python-modules`. However, if people are only expected it as a cli application (e.g.  `nimbo ...`). Then it should probably be packaged outside of `python-modules`. Going the application route gives you more freedom in deviating (can pin packages to other versions) from what is in `python-modules`.

Reply:
understood, I'll try moving it out of python-modules now

Reply:
Attempted in https://github.com/NixOS/nixpkgs/commit/1893b590516df977c15b3120ac82f7c7ae2cae80